src/js/BookReader/Mode1Up.js

```diff
@@ -12,144 +12,219 @@ export class Mode1Up {
 
+    // This calls drawLeafs
     this.resizePageView();
+
     this.br.jumpToIndex(startLeaf);
     this.br.updateBrClasses();
   }
 
-  drawLeafs() {
-    const { book } = this;
-    const containerHeight = this.br.refs.$brContainer.height();
-    const containerWidth = this.br.refs.$brPageViewEl.width();
-    const scrollTop = this.br.refs.$brContainer.prop('scrollTop');
-    const scrollBottom = scrollTop + containerHeight;
+  /**
+   * Get the number of pixels required to display the given inches with the given reduce
+   * @param {number} inches
+   **/
+  physicalInchesToDisplayPixels(inches, reduce = this.realWorldReduce, screenDPI = this.screenDPI) {
+    return inches * screenDPI / reduce;
+  }
 
-    const indicesToDisplay = [];
+  /** Iterate over pages, augmented with their top/bottom bounds */
+  * pagesWithBounds(reduce = this.realWorldReduce, leafSpacing = this.LEAF_SPACING_IN) {
     let leafTop = 0;
     let leafBottom = 0;
 
-    for (const page of book.pagesIterator({ combineConsecutiveUnviewables: true })) {
-      const height = Math.floor(page.height / this.br.reduce);
+    for (const page of this.book.pagesIterator({ combineConsecutiveUnviewables: true })) {
+      const height = this.physicalInchesToDisplayPixels(page.heightInches, reduce);
       leafBottom += height;
-      const topInView = (leafTop >= scrollTop) && (leafTop <= scrollBottom);
-      const bottomInView = (leafBottom >= scrollTop) && (leafBottom <= scrollBottom);
-      const middleInView = (leafTop <= scrollTop) && (leafBottom >= scrollBottom);
-      if (topInView || bottomInView || middleInView) {
-        indicesToDisplay.push(page.index);
-      }
-      leafTop += height + 10;
-      leafBottom += 10;
-    }
-
-    // Based of the pages displayed in the view we set the current index
-    // $$$ we should consider the page in the center of the view to be the current one
-    let firstIndexToDraw = indicesToDisplay[0];
-    this.br.updateFirstIndex(firstIndexToDraw);
-
-    // if zoomed out, also draw prev/next pages
-    if (this.br.reduce > 1) {
-      const prev = book.getPage(firstIndexToDraw).findPrev({ combineConsecutiveUnviewables: true });
-      if (prev) indicesToDisplay.unshift(firstIndexToDraw = prev.index);
-
-      const lastIndexToDraw = indicesToDisplay[indicesToDisplay.length - 1];
-      const next = book.getPage(lastIndexToDraw).findNext({ combineConsecutiveUnviewables: true });
-      if (next) indicesToDisplay.push(next.index);
+      yield { page, top: leafTop, bottom: leafBottom };
+      leafTop += height + this.physicalInchesToDisplayPixels(leafSpacing, reduce);
+      leafBottom += this.physicalInchesToDisplayPixels(leafSpacing, reduce);
     }
+  }
 
-    const BRpageViewEl = this.br.refs.$brPageViewEl.get(0);
-    leafTop = 0;
+  /**
+   * @param {{ top: number; bottom: number; }} bound1
+   * @param {{ top: number; bottom: number; }} bound2
+   */
+  static boundsIntersect(bound1, bound2) {
```

Comment:
suggestion: update jsdoc to reflect return value

Reply:
Fixed